File: desafio-02/hoklifter/python/primos.py
'''
Escreva um programa para listar todos os números primos entre 1 e 10000.
'''

def n_prime(num):
    '''Validador de números primos.'''
    for value in range(2, num):
        if num % value == 0:
            return False
    return True

def print_upto():
    '''Imprimir os números primos.'''
    up_to = 10000
    for value in range(2, up_to+1):
        if n_prime(value):
            print(value)
    while True:
        print_upto()
        break

File: desafio-02/joaoVitor-amaro/Primos.java
public class Primos {
    public static boolean isPrimo(int num) {
        if(num < 2) {
            return false;
        }
        for(int i = 2; i < num; i++) {
            if(num % i == 0) {
                return false;
            }
        }
        return true;
    }
    public static void main(String[] args) {
        int limit = 10000;
        for (int i = 1; i < limit; i++) {
            if(Primos.isPrimo(i)) {
                System.out.println(i);
            }
        }
    }
}
